Tools/Scripts/generate-compile-commands

```diff
@@ -45,6 +45,7 @@ parser = argparse.ArgumentParser(description='Generate compile_commands.json',
                   """)
 
 parser.add_argument('built_products_dir', help='path to the build directory containing generated compile commands (ex: WebKitBuild/Release)')
+parser.add_argument('--delete-invalid', action='store_true', help='Delete any invalid json file')
```

Comment:
do you usually need to say the `--` here? also, unfortunate that it mismatches the other argument in `_` vs. `-`, but yours seems more correct to me.

Reply:
`--` denotes an optional argument, without it it would be a compulsory "positional" argument. https://docs.python.org/3/library/argparse.html#name-or-flags

But yes, there's a discrepancy between `_` and `-`, and I also believe dashes `-` to be more correct or at least traditional. Should I fix that? (In a separate patch, good for me as a newbie!)

Reply:
Ah, got it!

And yeah, maybe see if you can fix `built-products-dir` separately :)